### ISSUE TITLE
Various issues starting with black screen.

### ISSUE THREAD

**Issue by devanteweary** [JonathanTreffler/backblaze-personal-wine-container#102]
Hey @JonathanTreffler ! Thanks for updating this!

A few days ago I checked your container (i.e. ran the WebUI on my Unraid Docker container) and the screen was completely black. Even after restart.

So I used Appdata Backup/Restore to restore to a few days ago and when I launched it again, it acted as if I hadn't backed up since December.
I also noticed the settings window is a bit different with the tabs all in a row on top.

Then it kinda... seemed like it was backing up starting in the right place.
Then this morning I got a Safety error which I guess is when your backup data log file shrunk.

Then it kinda... started backing up again.

So I clicked on the upper-left icon > Check for update and saw there's an update.
Clicking Download does nothing.

Checking logs show there are a bunch of "xdg-open: no method available for opening" and then some URL.
One being "secure.backblaze.com/update.htm?locale=en_US"

Lastly, a few days ago I was able to run "wine explorer" from the console to launch the file explorer inside the container.
And xkill to close it.

Now if I try that command, I get "wine: '/config/wine' is not owned by you"

So basically got a few little issues going!

Right now it looks like it's backing up correctly, but it seems something is wrong somewhere with all these issues.
Sorry it's kind of a bunch of random issues!

**Comment by JonathanTreffler** [comment]
You CANNOT under any circumstances roll back the bzdata files, which you did using Appdata Restore. Backblaze does not like that and the entire application relies on those files only ever growing as new stuff gets added.
You might have bricked you backup with this, so not just your local Backblaze installation, but also your stuff on the Backup servers. The only data I would trust right now in your situation is the files on your storage array, nothing that is restored from Backblaze.
But MAYBE you can still fix it by doing it the correct way: If your installation is broken you need to reset the docker (by changing all the volume paths the docker has to new empty directories or clearing out the existing ones), set the docker up from scratch (see the README) and use the inherit backup method of Backblaze (see: https://help.backblaze.com/hc/en-us/articles/217666358-Inherit-Backup-State), that way the newest bzdata files are downloaded from the Backup server.
The black screen issue is an issue of this docker that we are still figuring out, the rest of the issues is just how Backblaze works !

**Comment by devanteweary** [comment]
Doh I shoulda just left it! ha

Let me ask this... since it's as of now backing up a bunch of stuff I downloaded yesterday, is there a way to test if my backup is corrupt or not? 

My thinking is even if I do the inherit backup state, now it has bzdata files with the old "corrupt" backup and the new backups since yesterday.

Should I just start from scratch? (i.e. delete backup from web login > Reinstall Backblaze-Personal container from scratch?

**Comment by JonathanTreffler** [comment]
I think the general thing to remember is, that this docker creates a virtual Windows "computer" (wine instance) at the very first start, and it determines if the current start is the first start by whether the config folder is empty or not.
So if you want Backblaze to see this docker as a new computer on your account you need to create a new config folder and run the docker mapped to it. Then it creates a new computer and installs backblaze.

This is important, as you cannot inherit the backup from the same computer you are currently interacting backblaze with, because why would you want to do that, it would not change anything. So you need to reset the docker first, then it will show up as a new computer on your Backblaze account, with the previous computer showing as offline. Then you can either start from scratch by transferring your license before the trial on the new computer expires or you can use the inherit the old Backup to your new "computer" effectively merging them, so your new installation will start where it left off with the old one.

>  is there a way to test if my backup is corrupt or not?

You can test restore some newly uploaded file through the backblaze website and generate the SHA hash of it and of the same file on your storage array and compare it. But what you really need to look out for is any more errors from backblaze. If from here on out it will sometimes give you error messages I would not trust that backup anymore.

Sometimes it helps to think of the equivalent action, that you would have taken on a real normal computer to get the result with this virtual computer. Essentially rolling back to an older config directory is the same on this virtual computer as it would be to restore the whole C drive to an older backup on a real computer. Now that is something a normal Backblaze customer, installing through the normal install process without this docker might do, so maybe Backblaze has thought about this scenario and the program can actually deal with it, but I don't know.

**Comment by devanteweary** [comment]
Awesome explanation and really appreciate your dedication and willingness to help!
